File: 16subsetsums/py/solution.py
import sys

s = int(sys.stdin.readline())

numbers = [int(x) for x in sys.stdin.readline().strip().split(' ')]

sums = [0] * (s + 1)
sums[0] = True

for number in numbers:
    for current_sum in range(s, -1, -1):
        if not sums[current_sum]:
            continue

        new_sum = current_sum + number
        if new_sum > s:
            continue

        sums[new_sum] = True

print("yes" if sums[s] else "no") 

File: 16subsetsums/js/solution.js
let gets = this.gets || require('readline-sync').question;
let print = this.print || console.log;

const s = +gets()
const numbers = gets().split(' ').map(Number);

let sum = 0;
for (const number of numbers) {
    sum += number;
}

const sums = Array.from({
        length: sum + 1,
    })
    .fill(false);

sums[0] = true;

for (const number of numbers) {
    for (let sum = sums.length - 1; sum >= 0; sum -= 1) {
        if (sums[sum] > 0) {
            const newSum = sum + number;
            sums[newSum] = true;
        }
    }
}
print(sums[s]);
